File: src/lib/ipc/tracker.ts
import { invoke } from '@tauri-apps/api/core';
import type { TimerState } from '../types/timer';

export async function tracker_start(): Promise<TimerState> {
  return invoke('tracker_start');
}

export async function tracker_stop(): Promise<TimerState> {
  return invoke('tracker_stop');
}

export async function tracker_get_status(): Promise<TimerState> {
  return invoke('tracker_get_status');
}

File: src-tauri/src/controllers/tracker.rs
use crate::services::orchestrator::Orchestrator;
use crate::types::state::TimerState;
use std::sync::Arc;

#[tauri::command]
pub fn tracker_start(
    orchestrator: tauri::State<'_, Arc<Orchestrator>>,
) -> Result<TimerState, String> {
    orchestrator.handle_user_start()?;
    orchestrator.session.build_timer_state()
}

#[tauri::command]
pub fn tracker_stop(
    orchestrator: tauri::State<'_, Arc<Orchestrator>>,
) -> Result<TimerState, String> {
    orchestrator.handle_user_stop()?;
    orchestrator.session.build_timer_state()
}

#[tauri::command]
pub fn tracker_get_status(
    orchestrator: tauri::State<'_, Arc<Orchestrator>>,
) -> Result<TimerState, String> {
    orchestrator.session.build_timer_state()
}
